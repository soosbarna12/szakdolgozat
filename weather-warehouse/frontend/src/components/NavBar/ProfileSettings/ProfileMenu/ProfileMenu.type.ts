export interface ProfileMenuProps {
    anchorElUser: null | HTMLElement;
    isLightTheme: boolean;
    handleSetLightTheme: (isLightTheme: boolean) => void;
    handleCloseMenu: (event?: React.MouseEvent<HTMLElement>) => void;
}
