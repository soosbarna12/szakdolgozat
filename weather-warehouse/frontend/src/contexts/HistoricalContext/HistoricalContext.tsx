import React, { createContext, useEffect, useState } from "react";
import { LOCAL_STORAGE_TEMPERATURE_SCALE } from "../../consts/temperatureScale.const";
import { TemperatureScale } from "../../types/temperatureScale.type";
import { HistoricalContextType, HistoricalLocationData, HistoricalPageData, HistoricalProviderProps } from "./HistoricalContext.type";


export const HistoricalContext = createContext<HistoricalContextType>({
  location: { name: "", lat: 0, lon: 0 },
  setLocation: () => { },
  historicalPageData: [] as any,
  setHistoricalPageData: () => { },
  temperatureScale: TemperatureScale.Celsius,
  setTemperatureScale: () => { },
});

export const HistoricalLocationProvider = ({ children }: HistoricalProviderProps) => {

  const [location, setLocation] = useState<HistoricalLocationData>(() => {
    const storedLocation = localStorage.getItem("historicalLocation");
    const defaultLocation = { name: "", lat: 0, lon: 0 };
    try {
      return storedLocation ? JSON.parse(storedLocation) : defaultLocation;
    } catch (error) {
      console.error("Failed to parse location from localStorage:", error);
      return defaultLocation;
    }
  });

  const [historicalPageData, setHistoricalPageData] = useState<HistoricalPageData[]>(() => {
    const storedData = localStorage.getItem("historicalPageData");
    try {
      return storedData ? JSON.parse(storedData) : [];
    } catch (error) {
      console.error("Failed to parse historical data from localStorage:", error);
      return [];
    }
  });

  const [temperatureScale, setTemperatureScale] = useState<TemperatureScale | string>(() => {
    const storedTemperatureScale = localStorage.getItem(LOCAL_STORAGE_TEMPERATURE_SCALE);
    try {
      return storedTemperatureScale ? JSON.parse(storedTemperatureScale) : TemperatureScale.Celsius;
    } catch (error) {
      console.error("Failed to parse temperature scale from localStorage:", error);
      return TemperatureScale.Celsius;
    }
  });


  useEffect(() => {
    if (historicalPageData) {
      localStorage.setItem("historicalPageData", JSON.stringify(historicalPageData));
    }
  }, [historicalPageData]);

  useEffect(() => {
    localStorage.setItem("historicalLocation", JSON.stringify(location));
  }, [location]);

  useEffect(() => {
    localStorage.setItem(LOCAL_STORAGE_TEMPERATURE_SCALE, JSON.stringify(temperatureScale));
  }, [temperatureScale]);


  return (
    <HistoricalContext.Provider value={{ location, setLocation, historicalPageData, setHistoricalPageData, temperatureScale, setTemperatureScale }}>
      {children}
    </HistoricalContext.Provider>
  );
};
