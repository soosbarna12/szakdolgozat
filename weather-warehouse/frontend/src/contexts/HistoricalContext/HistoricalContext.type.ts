import { Dispatch, ReactNode, SetStateAction } from "react";
import { HistoricalDataTable } from "../../types/historicalDataTable.type";
import { TemperatureScale } from "../../types/temperatureScale.type";

export interface HistoricalLocationData {
  name: string;
  country?: string;
  lat: number;
  lon: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-object-type
export interface HistoricalPageData extends HistoricalDataTable {}

export interface HistoricalContextType {
  location: HistoricalLocationData;
  historicalPageData: any [];
  temperatureScale: TemperatureScale | string;
  setLocation: Dispatch<SetStateAction<HistoricalLocationData>>;
  setHistoricalPageData: Dispatch<SetStateAction<HistoricalPageData[]>>;
  setTemperatureScale: Dispatch<SetStateAction<TemperatureScale | string>>;
}

export interface HistoricalProviderProps {
  children: ReactNode;
}
