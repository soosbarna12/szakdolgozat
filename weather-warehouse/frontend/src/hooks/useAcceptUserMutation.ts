import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useAcceptUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.post(`/user/accept`, { id });
      return response.data;
    },
    onSuccess: () => {
      showAlert("User accepted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to accept user", "error");
    },
  });

  return mutation;
}
