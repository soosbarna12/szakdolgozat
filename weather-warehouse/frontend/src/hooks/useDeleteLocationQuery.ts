import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";

export function useDeleteLocationQuery() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (userLocationID: number) => {
      const response = await axios.delete(`/user/deleteLocation`, { params: { userLocationID } });
      return response.data;
    },
    onSuccess: () => {
      showAlert("Location deleted successfully", "success");
    },
    onError: () => {
      showAlert("Failed to delete location", "error");
    },
  });

  return mutation;
}