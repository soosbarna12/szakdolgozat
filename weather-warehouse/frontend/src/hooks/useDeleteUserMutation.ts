import { useMutation } from "@tanstack/react-query";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useDeleteUserMutation() {
  const { showAlert } = useAlert();

  const mutation = useMutation({
    mutationFn: async (id: number) => {
      const response = await axios.delete(`/user/delete`, { params: { id } });
      return response.data;
    },
    onSuccess: () => {
      showAlert('User deleted successfully', 'success');
    },
    onError: () => {
      showAlert('Failed to delete user', 'error');
    },
  });

  return mutation;
}
