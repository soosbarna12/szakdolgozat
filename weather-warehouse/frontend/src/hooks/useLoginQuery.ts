import { useQuery } from "@tanstack/react-query";
import { useEffect } from "react";
import { useAlert } from "../utils/AlertContext";
import axios from "../utils/axiosConfig";


export function useLoginQuery(username: string, password: string) {
  const { showAlert } = useAlert();
  
  const { data: loginData, error, isLoading, isSuccess, refetch } = useQuery({
    queryKey: ["login"],
    queryFn: async () => {
      const response = await axios.post(`/api/user/login`, {
          username,
          password
        });

      // store token (role is included in the token, getIsAdmin function will decode the token and check if the user is admin)
      localStorage.setItem('token', response.data.token);
      return response.data;
    },
      refetchOnWindowFocus: false,
      retry: false,
      enabled: false, // Disable automatic refetching, we will call refetch() manually -> to use this query when clicking the login button
  });

  useEffect(() => {
    if (isSuccess) {
      showAlert("Logged in successfully", "success");
    }
  }, [isSuccess]);

  useEffect(() => {
    if (error) {
      showAlert("Login failed", "error");
    }
  }, [error]);

  return { loginData, error, isLoading, isSuccess, refetch };
}
