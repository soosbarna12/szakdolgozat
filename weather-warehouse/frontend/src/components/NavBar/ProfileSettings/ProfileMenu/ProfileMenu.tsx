import DarkModeIcon from '@mui/icons-material/DarkMode';
import LightModeIcon from '@mui/icons-material/LightMode';
import { Box, Divider, Typography } from "@mui/material";
import React, { useContext, useEffect, useState } from "react";
import { HistoricalContext } from '../../../../contexts/HistoricalContext/HistoricalContext';
import { StyledToggleButton, StyledToggleButtonGroup } from '../../../../stlyes/button.style';
import { StyledMenu, StyledMenuItem } from '../../../../stlyes/common.style';
import { TemperatureScale } from '../../../../types/temperatureScale.type';
import { Theme } from '../../../../types/theme.type';
import { LoginForm } from '../../../AuthenticationForms/LoginForm/LoginForm';
import { ProfileMenuProps } from './ProfileMenu.type';


export function ProfileMenu(props: Readonly<ProfileMenuProps>) {
  const { anchorElUser, isLightTheme, handleSetLightTheme, handleCloseMenu } = props;
  const [openLogin, setOpenLogin] = useState(false);
  const [loggedIn, setLoggedIn] = useState(!!localStorage.getItem("token"));
  const { temperatureScale, setTemperatureScale } = useContext(HistoricalContext);
  const { setLocation, setHistoricalPageData } = useContext(HistoricalContext);

  useEffect(() => {
    setLoggedIn(!!localStorage.getItem("token"));
  }, [anchorElUser]);

  async function handleLoginClick() {
    setOpenLogin(true);
  }

  const handleLogout = () => {
    localStorage.removeItem("token");
    setLocation({ name: "", lat: 0, lon: 0 });
    setHistoricalPageData([]);
    setLoggedIn(false);
    handleCloseMenu();
    window.location.reload();
  }

  function handleCelsiusScale() {
    setTemperatureScale(JSON.stringify(TemperatureScale.Celsius));
  }

  function handleFahrenheitScale() {
    setTemperatureScale(JSON.stringify(TemperatureScale.Fahrenheit));
  }

  function handleLightModeIconClick() {
    handleSetLightTheme(true);
  }

  function handleDarkModeIconClick() {
    handleSetLightTheme(false);
  }

  function handleLoginSuccess() {
    setLoggedIn(true);
  }

  function handleCloseLoginForm() {
    setOpenLogin(false);
    handleCloseMenu();
  }

  function renderTemperatureScaleToggle() {
    return (
      <Box>
        <StyledToggleButtonGroup>
          <StyledToggleButton
            value={TemperatureScale.Celsius}
            onClick={handleCelsiusScale}
            disabled={temperatureScale === TemperatureScale.Celsius}>
            °C
          </StyledToggleButton>
          <StyledToggleButton
            value={TemperatureScale.Fahrenheit}
            onClick={handleFahrenheitScale}
            disabled={temperatureScale === TemperatureScale.Fahrenheit}>
            °F
          </StyledToggleButton>
        </StyledToggleButtonGroup>
      </Box>
    );
  }

  function renderThemeToggle() {
    return (
      <Box sx={{ marginTop: "6px" }}>
        <StyledToggleButtonGroup>
          <StyledToggleButton
            value={Theme.Light}
            onClick={handleLightModeIconClick}
            data-testid="lightModeButton"
            disabled={isLightTheme}>
            <LightModeIcon />
          </StyledToggleButton>
          <StyledToggleButton
            value={Theme.Dark}
            onClick={handleDarkModeIconClick}
            disabled={!isLightTheme}
            data-testid="darkModeButton">
            <DarkModeIcon />
          </StyledToggleButton>
        </StyledToggleButtonGroup>
      </Box>
    );
  }

  function renderLoggedOutMenuItem() {
    if (!loggedIn) {
      return (
        <StyledMenuItem onClick={handleLoginClick}>
          <Typography sx={{ textAlign: "center" }}>Log In</Typography>
        </StyledMenuItem>
      )
    }
  }

  function renderLoggedInMenuItems() {
    if (loggedIn) {
      return (
        <div>
          <StyledMenuItem onClick={handleLogout}>
            <Typography sx={{ textAlign: "center" }}>Log Out</Typography>
          </StyledMenuItem>
        </div>
      )
    }
  }

  return (
    <>
      <StyledMenu
        sx={{ marginTop: "53px", padding: "0", borderRadius: "20px" }}
        id="menu-appbar"
        anchorEl={anchorElUser}
        anchorOrigin={{ vertical: "top", horizontal: "right" }}
        keepMounted
        transformOrigin={{ vertical: "top", horizontal: "right" }}
        open={Boolean(anchorElUser)}
        onClose={() => handleCloseMenu()}>

        {renderLoggedOutMenuItem()}
        {renderLoggedInMenuItems()}

        <Divider style={{ margin: "10px" }} />

        {renderTemperatureScaleToggle()}
        {renderThemeToggle()}

      </StyledMenu>
      <LoginForm
        open={openLogin}
        onClose={handleCloseLoginForm}
        onLoginSuccess={handleLoginSuccess}
      />
    </>
  )
}
