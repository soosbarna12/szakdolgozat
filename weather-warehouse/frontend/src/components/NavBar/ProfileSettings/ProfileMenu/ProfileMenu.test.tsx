import '@testing-library/jest-dom';
import { fireEvent, screen } from '@testing-library/react';
import React from 'react';
import { HistoricalContext } from '../../../../contexts/HistoricalContext/HistoricalContext';
import { useLoginQuery } from '../../../../hooks/useLoginQuery';
import { renderWithQueryClient } from '../../../../utils/test/renderWithQueryClient';
import { ProfileMenu } from './ProfileMenu';

jest.mock('../../../../hooks/useLoginQuery', () => ({
  useLoginQuery: jest.fn(),
}));

jest.mock('../../../AuthenticationForms/LoginForm/LoginForm', () => ({
  LoginForm: ({ open, onClose, onLoginSuccess }: any) => (
    <div data-testid="loginForm">
      {open && (
        <div>
          <button data-testid="loginSuccessButton" onClick={onLoginSuccess}>
            Login Success
          </button>
          <button data-testid="closeLoginFormButton" onClick={onClose}>
            Close Login Form
          </button>
        </div>
      )}
    </div>
  ),
}));

describe('NavBar/ProfileSettings/ProfileMenu', () => {
  const mockHandleSetLightTheme = jest.fn();
  const mockHandleCloseMenu = jest.fn();
  const mockSetTemperatureScale = jest.fn();
  const mockSetLocation = jest.fn();
  const mockSetHistoricalPageData = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    (useLoginQuery as jest.Mock).mockImplementation(() => ({
      loginData: null,
      error: null,
      isLoading: false,
      isSuccess: true,
      refetch: jest.fn(),
    }));
  });

  beforeAll(() => {
    Object.defineProperty(window, 'location', {
      value: {
        ...window.location,
        reload: jest.fn(),
      },
      writable: true,
    });
  });

  const renderComponent = (anchorElUser: HTMLElement | null, isLightTheme = true) => {
    return renderWithQueryClient(
      <HistoricalContext.Provider
        value={{
          temperatureScale: 'Celsius',
          setTemperatureScale: mockSetTemperatureScale,
          setLocation: mockSetLocation,
          setHistoricalPageData: mockSetHistoricalPageData,
        } as any}
      >
        <ProfileMenu
          anchorElUser={anchorElUser}
          isLightTheme={isLightTheme}
          handleSetLightTheme={mockHandleSetLightTheme}
          handleCloseMenu={mockHandleCloseMenu}
        />
      </HistoricalContext.Provider>
    );
  };

  it('matches ProfileMenu component snapshot', () => {
    const component = renderComponent(document.createElement('div'));
    expect(component).toMatchSnapshot();
  });

  it('renders login menu item when user is not logged in', () => {
    renderComponent(document.createElement('div'), true);

    const loginMenuItem = screen.getByText('Log In');
    expect(loginMenuItem).toBeInTheDocument();

    fireEvent.click(loginMenuItem);

    expect(screen.getByTestId('loginForm')).toBeInTheDocument();
  });

  it('renders logout menu item when user is logged in', () => {
    localStorage.setItem('token', 'mockToken');
    renderComponent(document.createElement('div'), true);

    const logoutMenuItem = screen.getByText('Log Out');
    expect(logoutMenuItem).toBeInTheDocument();

    fireEvent.click(logoutMenuItem);

    expect(mockSetLocation).toHaveBeenCalledWith({ name: '', lat: 0, lon: 0 });
    expect(mockSetHistoricalPageData).toHaveBeenCalledWith([]);
    expect(mockHandleCloseMenu).toHaveBeenCalled();
    expect(localStorage.getItem('token')).toBeNull();
  });

  it('toggles temperature scale to Fahrenheit', () => {
    renderComponent(document.createElement('div'));

    const fahrenheitButton = screen.getByText('°F');
    fireEvent.click(fahrenheitButton);

    expect(mockSetTemperatureScale).toHaveBeenCalledWith('fahrenheit');
  });

  it('toggles temperature scale to Celsius', () => {
    renderComponent(document.createElement('div'));

    const celsiusButton = screen.getByText('°C');
    fireEvent.click(celsiusButton);

    expect(mockSetTemperatureScale).toHaveBeenCalledWith('celsius');
  });

  it('toggles theme to dark mode', () => {
    renderComponent(document.createElement('div'), true);

    const darkModeButton = screen.getByTestId('darkModeButton');
    fireEvent.click(darkModeButton);

    expect(mockHandleSetLightTheme).toHaveBeenCalledWith(false);
  });

  it('toggles theme to light mode', () => {
    renderComponent(document.createElement('div'), false);

    const lightModeButton = screen.getByTestId('lightModeButton');
    fireEvent.click(lightModeButton);

    expect(mockHandleSetLightTheme).toHaveBeenCalledWith(true);
  });
});