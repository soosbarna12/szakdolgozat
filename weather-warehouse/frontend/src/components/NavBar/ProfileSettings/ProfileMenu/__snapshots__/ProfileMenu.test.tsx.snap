// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar/ProfileSettings/ProfileMenu matches ProfileMenu component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        data-testid="loginForm"
      />
    </div>
    <div
      class="MuiPopover-root MuiMenu-root MuiModal-root css-1x3suky-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-vl6bb4-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper css-1tktgsa-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="--Paper-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12); opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-1toxriw-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-focusVisible MuiMenuItem-root MuiMenuItem-gutters css-nl63xq-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1xrrpwr-MuiTypography-root"
            >
              Log In
            </p>
          </li>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            style="margin: 10px;"
          />
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-1kk3p74-MuiToggleButtonGroup-root"
              role="group"
            >
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-firstButton css-1mup24r-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="celsius"
              >
                °C
              </button>
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-lastButton css-1mup24r-MuiButtonBase-root-MuiToggleButton-root"
                tabindex="0"
                type="button"
                value="fahrenheit"
              >
                °F
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-19ug521"
          >
            <div
              class="MuiToggleButtonGroup-root MuiToggleButtonGroup-horizontal css-1kk3p74-MuiToggleButtonGroup-root"
              role="group"
            >
              <button
                aria-pressed="false"
                class="MuiButtonBase-root Mui-disabled MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root Mui-disabled MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-firstButton css-1mup24r-MuiButtonBase-root-MuiToggleButton-root"
                data-testid="lightModeButton"
                disabled=""
                tabindex="-1"
                type="button"
                value="lightTheme"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="LightModeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5M2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1m18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1M11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1m0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1M5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0z"
                  />
                </svg>
              </button>
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-lastButton css-1mup24r-MuiButtonBase-root-MuiToggleButton-root"
                data-testid="darkModeButton"
                tabindex="0"
                type="button"
                value="darkTheme"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="DarkModeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1"
                  />
                </svg>
              </button>
            </div>
          </div>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      data-testid="loginForm"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
